syntax = "proto3";

package payment;

service PaymentService {
  rpc CheckoutPrepare (TxnRequest) returns (Status);
  rpc CheckoutCommit (TxnRequest) returns (UserDataResponse);
  rpc CheckoutRollback (TxnIDRequest) returns (Status);
}

message UserDataResponse {
  string user_id = 1;
  int32 credit = 2;
}

message TxnRequest {
  string user_id = 1;
  string transaction_id = 2;
  int32 amount = 3;
}

message TxnIDRequest {
  string user_id = 1;
  string transaction_id = 2;
}

message Status {
  string msg = 1;
  int32 code = 2;
}
